### DISCUSSION TITLE
How to automatically update Adnauseam on Chrome/Mac ?

### DISCUSSION THREAD

**Discussion by davids74** [dhowe/AdNauseam#2480] [General]
Since 2022 Adnauseam ad blocker for Chrome has to be loaded from and stored in a separate folder somewhere on a hard drive.
Downloading the latest release as a .zip file and extracting copying over the previous version is mildly tedious.

I wonder if there is a OSX terminal command that could be run to compare the local version to online one ?
I've looked around and it does not seem to be described anywhere...

**Comment by davids74** [comment]
I tried
`/Downloads/gh_2.39.2_macOS_amd64/bin/gh repo clone dhowe/AdNauseam` 
but is says
```
dyld: Symbol not found: _clock_gettime
  Referenced from: /Volumes/Docs/david/Downloads/gh_2.39.2_macOS_amd64/bin/gh
  Expected in: flat namespace

Trace/BPT trap: 5
```
I did not install it via brew because brew asks for xcode which does not install on my old computer.

**Comment by mneunomne** [comment]
Hi @davids74, that's a good idea yes.

I made this simple shell script that compares the online latest non-prerelease with the local one. I guess you can set it in some sort of auto-run?

```sh
#!/bin/bash

# Path to the adnauseam.chromium directory
EXTENSION_DIR="adnauseam.chromium"

# Path to the manifest.json file
MANIFEST_FILE="$EXTENSION_DIR/manifest.json"

# Check if the directory exists
if [ -d "$EXTENSION_DIR" ]; then
    # Get the current version from the manifest file
    CURRENT_VERSION=$(jq -r .version "$MANIFEST_FILE")
    CURRENT_VERSION="v$CURRENT_VERSION"
    echo "Current version: $CURRENT_VERSION"

    # Replace URL with the actual URL you want to use
    URL="https://api.github.com/repos/dhowe/AdNauseam/releases/latest"

    # Get the latest release information using curl and jq
    LATEST_RELEASE=$(curl -s "$URL" | jq -r '.tag_name')

    # Check if the latest version is different from the current version
    if [ "$LATEST_RELEASE" != "$CURRENT_VERSION" ]; then
        echo "Updating AdNauseam extension..."

        # Download and extract the latest release
        curl -L -o "$EXTENSION_DIR.zip" "https://github.com/dhowe/AdNauseam/releases/download/$LATEST_RELEASE/adnauseam.chromium.zip"
        unzip -q "$EXTENSION_DIR.zip" -d "./"
        rm "$EXTENSION_DIR.zip"

        echo "Update complete. Installed version: $LATEST_RELEASE"
    else
        echo "AdNauseam is already up to date. Installed version: $CURRENT_VERSION"
    fi
else
    echo "AdNauseam extension directory not found. Please make sure it is installed."
fi
```

**Comment by davids74** [comment]
I copy pasted it into terminal and got response
AdNauseam extension directory not found. Please make sure it is installed.

**Comment by davids74** [comment]
I tried
Davids-iMac:~ david$ cd $EXTENSION_DIR
-bash: cd: adnauseam.chromium: No such file or directory
Davids-iMac:~ david$

**Comment by davids74** [comment]
I replaced extension dir with "/Volumes/Docs/david/Documents/adnauseam.chromium" and it did something

`Davids-iMac:~ david$ if [ -d "$EXTENSION_DIR" ]; then
>     # Get the current version from the manifest file
>     CURRENT_VERSION=$(jq -r .version "$MANIFEST_FILE")
>     CURRENT_VERSION="v$CURRENT_VERSION"
>     echo "Current version: $CURRENT_VERSION"
> 
>     # Replace URL with the actual URL you want to use
>     URL="https://api.github.com/repos/dhowe/AdNauseam/releases/latest"
> 
>     # Get the latest release information using curl and jq
>     LATEST_RELEASE=$(curl -s "$URL" | jq -r '.tag_name')
> 
>     # Check if the latest version is different from the current version
>     if [ "$LATEST_RELEASE" != "$CURRENT_VERSION" ]; then
>         echo "Updating AdNauseam extension..."
> 
>         # Download and extract the latest release
>         curl -L -o "$EXTENSION_DIR.zip" "https://github.com/dhowe/AdNauseam/releases/download/$LATEST_RELEASE/adnauseam.chromium.zip"
>         unzip -q "$EXTENSION_DIR.zip" -d "./"
>         rm "$EXTENSION_DIR.zip"
> 
>         echo "Update complete. Installed version: $LATEST_RELEASE"
>     else
>         echo "AdNauseam is already up to date. Installed version: $CURRENT_VERSION"
>     fi
> else
>     echo "AdNauseam extension directory not found. Please make sure it is installed."
> fi
-bash: jq: command not found
Current version: v
-bash: jq: command not found
(23) Failed writing body
Updating AdNauseam extension...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     9  100     9    0     0     10      0 --:--:-- --:--:-- --:--:--    10
[/Volumes/Docs/david/Documents/adnauseam.chromium.zip]
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /Volumes/Docs/david/Documents/adnauseam.chromium.zip or
        /Volumes/Docs/david/Documents/adnauseam.chromium.zip.zip, and cannot find /Volumes/Docs/david/Documents/adnauseam.chromium.zip.ZIP, period.
Update complete. Installed version: 
Davids-iMac:~ david$ 
`

**Comment by mneunomne** [comment]
Ah yes, sorry I forgot to say that you need to be in the folder where your adnauseam.chromium is located...

Also you need `jq` installed: https://stackoverflow.com/questions/37668134/how-to-install-jq-on-mac-on-the-command-line

**Comment by davids74** [comment]
I try to run Xcode to install brew but it just hangs on verification. 
![Screen Shot 2024-01-15 at 13 14 57](https://github.com/dhowe/AdNauseam/assets/39235819/8bcca031-0ff9-4dc5-a88c-5e3671460ad7)

![Screen Shot 2024-01-15 at 13 19 30](https://github.com/dhowe/AdNauseam/assets/39235819/f16a4c75-a972-4c28-8626-1f04f18dccd5)

It seems strange that you need to download and install a 2GB app to download and install 2kb of code.

**Reply by dhowe** [reply]
it sure does... talk to apple

**Reply by mneunomne** [reply]
True you are running an old version of Mac, sometimes it's hard to install things on terminal with them from my own experience...

**Comment by dhowe** [comment]
This is a difficult but important problem/feature, but I do think we need to maintain cross-platform compatibility, which includes, however unfortunately, windows...
